src/antsibull/build_changelog.py

```diff
@@ -204,64 +301,89 @@ def append_changelog_changes_ansible(
 
     release_summary = release_entry.changes.pop("release_summary", None)
     if release_summary:
-        builder.add_section("Release Summary", 1)
-        builder.add_raw_rst(t.cast(str, release_summary))
-        builder.add_raw_rst("")
+        section = renderer.add_section("Release Summary")
+        section.add_text(
+            t.cast(str, release_summary), text_format=TextFormat.RESTRUCTURED_TEXT
```

Comment:
If you annotate the declaration of `release_summary` with `release_summary: str | None` you should be able to get rid of the cast, but I guess it was there before.

Reply:
That results in mypy complaining:
```
src/antsibull/build_changelog.py:297: error: Incompatible types in assignment (expression has type "str | list[str] | None", variable has type "str | None")  [assignment]
```